rewrite-java/src/main/java/org/openrewrite/java/search/NotUsesType.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.search;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Value;
+import org.jspecify.annotations.Nullable;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class NotUsesType extends Recipe {
+
+    @Nullable
+    @Getter
+    private final String fullyQualifiedType;
+
+    @Nullable
+    private final Boolean includeImplicit;
+
+    @Override
+    public String getDisplayName() {
+        return "Check whether a type is not in use";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Inverse of `UsesType`.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        return Preconditions.not(new UsesType<>(fullyQualifiedType, includeImplicit));
+    }
```

Comment:
1. The fields here are likely not nullable, so remove the nullable annotations; that then also removes these IDE warnings.
![image](https://github.com/user-attachments/assets/b522c88e-5a5b-4436-8a7b-c27dc767c94c)

2. Also remove the `private final` as indicated by my IDE.

3. Let's rename class to `DoesNotUseType`, as `NotUsesType` sounds a bit awkward.

4. "Inverse of `UsesType`" isn't very descriptive in and of itself; instead let's describe the intended use as a precondition to invalidate classes using a particular type. e.g. do not run recipe X when type Y is used in a class.